src/ralph/virtual/management/commands/openstack_sync.py

```diff
@@ -706,8 +706,21 @@ def _cleanup(self):
         for project_id in (set(self.ralph_projects.keys()) - set(
             self.openstack_projects.keys())
         ):
-            self._delete_object(CloudProject.objects.get(
-                project_id=project_id))
+            cloud_project = CloudProject.objects.get(project_id=project_id)
+            children_count = cloud_project.children.count()
+            if children_count == 0:
+                self._delete_object(CloudProject.objects.get(
```

Comment:
Could you please add a log message with the extras I mentioned on L717 here as well?